tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java

```diff
@@ -190,6 +190,30 @@ public void addAllocationId(String jobName, long allocationRequestId) {
     LOG.info(String.format("Job %s with allocationRequestId %d", jobName, allocationRequestId));
   }
 
+  /**
+   * Get a TensorFlow task that hasn't been scheduled.
+   * In the absense of allocationRequestId, we are relying on the fact that each tensorflow job will
+   * have a distinct priority (Ensured in {@link Utils#parseContainerRequests(Configuration)}).
+   * @param priority the priority of the allocated container
+   * @return task to be assigned to this allocation
+   */
+  public synchronized TonyTask getAndInitMatchingTaskByPriority(int priority) {
+    for (Map.Entry<String, TensorFlowContainerRequest> entry : containerRequests.entrySet()) {
+      String jobName = entry.getKey();
+      if (entry.getValue().getPriority() != priority) {
+        LOG.debug("Ignoring jobname {" + jobName + "} as priority doesn't match");
```

Comment:
We need to continue; here right? We should be matching this container only with the tasks with the same priority.